### PR TITLE
PHP 8 compatibility

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change

Upgrade the plugin to be compatible with the least supportive PHP v5.3 up to the PHP v8.1

### Verification Process

Make sure the Github actions are passing. Also, test the full plugin functionalities. 

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Changelog Entry

> Security - Plugin made compatible up to PHP 8.1

### Credits

Props @faisal-alvi @dkotter @Sidsector9

### CODE REVIEW

File: .github/workflows/php-compatibility.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - develop
+
+jobs:
+  php_compatibility:
+    name: ${{ matrix.php }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        php: [ '5.6', '7.0', '7.4', '8.0', '8.1' ]
```
Comment (Sidsector9) [inline .github/workflows/php-compatibility.yml]: You can remove this matrix here and from the places where it is used. See this [Slack post](https://10up.slack.com/archives/C84M9AA06/p1648460113851159) for more details.

**Review by Sidsector9 — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          path: ${{ env.COMPOSER_CACHE }}
+          key: composer-${{ env.COMPOSER_VERSION }}-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            composer-${{ env.COMPOSER_VERSION }}-
+      - name: Set PHP version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.2'
+          coverage: none
+          tools: prestissimo, composer:v2
```
Comment (Sidsector9) [inline .github/workflows/lint.yml:38]: `prestissimo` is not necessary if you're using composer v2.
Read more here: https://github.com/hirak/prestissimo#announcement-composer-2-is-now-available

**Review by Sidsector9 — CHANGES_REQUESTED** [review]

File: .github/workflows/php-compatibility.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          key: composer-${{ env.COMPOSER_VERSION }}-${{ hashFiles('**/composer.lock') }}
+          restore-keys: |
+            composer-${{ env.COMPOSER_VERSION }}-
+
+      - name: Set PHP version
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: '7.2'
+          coverage: none
+          tools: prestissimo, composer:v2
```
Comment (Sidsector9) [inline .github/workflows/php-compatibility.yml]: `prestissimo` is unnecessary if we're using composer v2,
[Refer to this](https://github.com/hirak/prestissimo#announcement-composer-2-is-now-available) for more details.

**Review by Sidsector9 — CHANGES_REQUESTED** [review]

File: .github/workflows/lint.yml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - name: composer install
+        run: composer install --ignore-platform-reqs
+      - name: Check PHPCS standard
+        run: ./vendor/bin/phpcs -i
+      - name: PHPCS check
+        uses: chekalsky/phpcs-action@v1
```
Comment (Sidsector9) [inline .github/workflows/lint.yml]: We're already using [10up/phpcs-composer](https://github.com/10up/phpcs-composer) in the project. Would be great to keep tools consistent.

Refer [to this section](https://docs.google.com/document/d/1r6WSXn22bOYfotDd9oyqf_CPsPXBGK2M1TfKDeKQ7Ew/edit#heading=h.mhv3eyi68ryb) for more details.

**Review by Sidsector9 — CHANGES_REQUESTED** [review]

**Comment by faisal-alvi** [comment]
@Sidsector9 just a reminder, this is ready for review.

**Review by Sidsector9 — APPROVED** [review]

**Pull request merged**